apps/vercel/frontend/src/hooks/useFetchData/useFetchData.tsx

```diff
@@ -102,5 +102,30 @@ export const useFetchData = ({
     }
   };
 
-  return { validateToken, fetchProjects, fetchApiPaths };
+  const validateProjectEnv = async (currentSpaceId: string, projectId: string) => {
+    if (vercelClient && teamId)
+      try {
+        const isProjectSelectionValid = await vercelClient.validateProjectContentfulSpaceId(
+          currentSpaceId,
+          projectId,
+          teamId
+        );
+        if (!isProjectSelectionValid) throw new Error(errorTypes.INVALID_SPACE_ID);
+        else {
+          dispatchErrors({
+            type: errorsActions.RESET_PROJECT_SELECTION_ERRORS,
+          });
+        }
+      } catch (e) {
+        const err = e as Error;
+        if (err.message === errorTypes.INVALID_SPACE_ID) {
```

Comment:
Not a super big deal but I tend to like the pattern of creating a custom error type and then handling that error type:

```ts
// custom error class
class InvalidProjectSpaceId extends Error {}

// execution
if (!isProjectSelectionValid) throw new InvalidProjectSpaceId(`The current space ID '${currentSpaceId}' doesn't match the space id defined for the project`)

// handling
if (err instanceof InvalidProjectSpaceId) {
  // handle it
}
```

Even if you don't handle the message now it's just an easier way to do something with it in the future

Reply:
Yeah awesome suggestion! I will definitely note this for the future 👍